src/storage/src/hummock/compactor/shared_buffer_compact.rs

```diff
@@ -179,13 +183,28 @@ async fn compact_shared_buffer(
         }
         let compaction_executor = context.compaction_executor.clone();
         let multi_filter_key_extractor = multi_filter_key_extractor.clone();
-        let handle = compaction_executor.spawn(async move {
-            compactor
-                .run(
-                    MergeIterator::new(forward_iters),
-                    multi_filter_key_extractor,
+        let handle = compaction_executor.spawn({
+            static NEXT_SHARED_BUFFER_COMPACT_ID: LazyLock<AtomicUsize> =
+                LazyLock::new(|| AtomicUsize::new(0));
+            let tree_root = context.await_tree_reg.as_ref().map(|reg| {
+                let id = NEXT_SHARED_BUFFER_COMPACT_ID.fetch_add(1, Relaxed);
+                reg.write().register(
+                    format!("compact_shared_buffer/{}", id),
+                    format!(
+                        "Compact Shared Buffer: {:?}",
+                        payload.iter().map(|imm| imm.batch_id()).collect_vec()
```

Comment:
I am concerned that this will be too big when there are too many actors (>1k) in the CN. How about using epochs instead?